snap/quota/resources_test.go

```diff
@@ -169,3 +171,28 @@ func (s *resourcesTestSuite) TestQuotaChangeValidationPasses(c *C) {
 		c.Check(t.limits, DeepEquals, t.newLimits)
 	}
 }
+
+func (s *resourcesTestSuite) TestResourceCloneComplete(c *C) {
+	r := &quota.Resources{}
+	rv := reflect.ValueOf(r).Elem()
+	fieldPtrs := make([]uintptr, rv.NumField())
+	for i := 0; i < rv.NumField(); i++ {
+		fv := rv.Field(i)
+		ft := fv.Type()
+		nv := reflect.New(ft.Elem())
```

Comment:
I think this test assumes that the fields are always pointers to (or containers of) some type and that they're always initialized right? Is that a reasonable assumption in this context?

Reply:
That's how the structure is currently defined. I'm not sure why the Resources structure uses pointers instead of embedding the sub-structures directly, but should that change, we can update the test.

Reply:
Yeah, it uses pointers right now, I don't know the deeper reason, I assume because json will omit nil more easily than an empty struct?